src/OpenTelemetry.SpanProcessor.Baggage/BaggageSpanProcessor.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace OpenTelemetry.SpanProcessor.Baggage;
+
+/// <summary>
+/// Span processor that adds <see cref="Baggage"/> fields to every new span
+/// </summary>
+public class BaggageSpanProcessor : OpenTelemetry.<Activity>
```

Comment:
I'd suggest to keep this internal, and offer extension method like https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Extensions/TracerProviderBuilderExtensions.cs#L32